internal/dao/pgx/reservation_pgx.go

```diff
@@ -360,3 +362,17 @@ func (x *reservationDao) Delete(ctx context.Context, id int64) error {
 	}
 	return nil
 }
+
+func (x *reservationDao) Cleanup(ctx context.Context) error {
+	logger := zerolog.Ctx(ctx)
+	query := `DELETE FROM reservations WHERE created_at < now() - cast($1 as interval)`
+	reservationLifetime := config.Reservation.Lifetime.String()
+
+	tag, err := db.Pool.Exec(ctx, query, reservationLifetime)
+	if err != nil {
+		return fmt.Errorf("pgx error: %w", err)
+	}
+	logger.Trace().Msgf("Deleted %d reservation(s) older than %s", tag.RowsAffected(), reservationLifetime)
+
+	return nil
+}
```

Comment:
Note for myself: duration.String() returns "1d2h3s" format https://pkg.go.dev/maze.io/x/duration#Duration.String and postgres accepts the following https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-INTERVAL-INPUT